cdap-watchdog/src/main/java/co/cask/cdap/metrics/store/upgrade/DataMigration.java

```diff
@@ -0,0 +1,329 @@
+      .put("w", ImmutableList.of(Constants.Metrics.Tag.APP,
+                                 TYPE,
+                                 Constants.Metrics.Tag.WORKFLOW,
+                                 Constants.Metrics.Tag.INSTANCE_ID))
+      .build();
+
+  private final Map<String, String> metricNameToTagNameMapping =
+    ImmutableMap.<String, String>builder()
+      .put("store.reads", Constants.Metrics.Tag.DATASET)
+      .put("store.writes", Constants.Metrics.Tag.DATASET)
+      .put("store.ops", Constants.Metrics.Tag.DATASET)
+      .put("store.bytes", Constants.Metrics.Tag.DATASET)
+      .put("dataset.store.reads", Constants.Metrics.Tag.DATASET)
+      .put("dataset.store.writes", Constants.Metrics.Tag.DATASET)
+      .put("dataset.store.ops", Constants.Metrics.Tag.DATASET)
+      .put("dataset.store.bytes", Constants.Metrics.Tag.DATASET)
+      .put("dataset.size.mb", Constants.Metrics.Tag.DATASET)
+      .put("collect.events", Constants.Metrics.Tag.STREAM)
+      .put("collect.bytes", Constants.Metrics.Tag.STREAM)
+      .put("process.tuples.read", Constants.Metrics.Tag.FLOWLET_QUEUE)
+      .put("process.events.in", Constants.Metrics.Tag.FLOWLET_QUEUE)
+      .put("process.events.out", Constants.Metrics.Tag.FLOWLET_QUEUE)
+      .put("process.events.processed", Constants.Metrics.Tag.FLOWLET_QUEUE)
+      .build();
+
+  private final Map<String, Map<String, String>> mapOldSystemContextToNew =
+    ImmutableMap.<String, Map<String, String>>of(
+      "transactions", ImmutableMap.of(Constants.Metrics.Tag.NAMESPACE, Constants.SYSTEM_NAMESPACE,
+                                      Constants.Metrics.Tag.COMPONENT, "transactions"),
+      "-", ImmutableMap.of(Constants.Metrics.Tag.NAMESPACE, Constants.SYSTEM_NAMESPACE),
+
+      "gateway", ImmutableMap.of(Constants.Metrics.Tag.NAMESPACE, Constants.SYSTEM_NAMESPACE,
+                                 Constants.Metrics.Tag.COMPONENT, Constants.Gateway.METRICS_CONTEXT,
+                                 Constants.Metrics.Tag.HANDLER, Constants.Gateway.STREAM_HANDLER_NAME)
+    );
+  private final String entityTableName;
+  private final String metricsTableName;
+
+  public DataMigration(final CConfiguration cConf, final DatasetFramework dsFramework,
+                       DefaultMetricDatasetFactory factory) {
+
+    LOG.info("Initializing Data Migration.");
+    this.dsFramework = dsFramework;
+    this.entityTableName = cConf.get(MetricsConstants.ConfigKeys.ENTITY_TABLE_NAME,
+                                     UpgradeMetricsConstants.DEFAULT_ENTITY_TABLE_NAME);
+    this.metricsTableName = cConf.get(MetricsConstants.ConfigKeys.METRICS_TABLE_PREFIX,
+                                      UpgradeMetricsConstants.DEFAULT_METRICS_TABLE_PREFIX) + ".agg";
+    aggMetricStore = new DefaultMetricStore(factory, new int[]{Integer.MAX_VALUE});
+  }
+
+  public void migrateMetricsTable(Version cdapVersion) {
+    if (cdapVersion.getVersion().equals("2.6")) {
```

Comment:
not cool to check like that, but it's ok for now - since we are going to hook it up with detecting version and this will be changed